src/figuresClasses.ts

```diff
@@ -1,19 +1,80 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('Size of side should be greater than 0');
+    }
```

Comment:
you need to check that input arguments are bigger than 0 for each figure
instead of checking it in each class, let's create one separate function for this purpose, and let's call it for each figure